src/local-history/local-history-manager.ts

```diff
@@ -75,93 +72,117 @@ export class LocalHistoryManager {
         return `${year}-${month}-${day} ${hour}:${minute}:${second}`;
     }
 
+    /**
+     * Return the timestamp in the following format: 2020-04-06 10:25:50.123.
+     */
+    private getCurrentTime(): string {
+        let time = new Date();
+        time = new Date(time.getTime() - (time.getTimezoneOffset() * 60000));
+        return time.toISOString().substring(0, 23).replace(/[T]/g, ' ');
+    }
+
+    /**
+    * Load existing entries on activation from local file system.
+    */
+    public async loadHistoryForActiveEditor(hashedFolderPath: string): Promise<void> {
+
+        if (!vscode.workspace.workspaceFolders || vscode.workspace.workspaceFolders.length <= 0) {
+            return;
+        }
+
+        this.historyFilesForActiveEditor = [];
+        const pathInVscodeUri = vscode.Uri.file(hashedFolderPath);
+
+        try {
+            for (const [fileName, type] of await vscode.workspace.fs.readDirectory(pathInVscodeUri)) {
+                if (type === vscode.FileType.File) {
+                    const data: HistoryFileProperties = {
+                        fileName: fileName,
+                        timestamp: this.parseTimestamp(fileName),
+                        uri: path.join(hashedFolderPath, fileName),
+                    };
+                    this.historyFilesForActiveEditor.unshift(data);
+                }
+            }
+        } catch (err) {
+            vscode.window.showInformationMessage(err);
+        }
+    }
```

Comment:
[question]: Should we display the same error to the user, or handle it more gracefully by creating an error string specific to the case?